homeassistant/components/switch/rpi_rf.py

```diff
@@ -101,22 +105,23 @@ def is_on(self):
         """Return true if device is on."""
         return self._state
 
-    def _send_code(self, code, protocol, pulselength):
-        """Send the code with a specified pulselength."""
-        _LOGGER.info("Sending code: %s", code)
-        res = self._rfdevice.tx_code(code, protocol, pulselength)
-        if not res:
-            _LOGGER.error("Sending code %s failed", code)
-        return res
+    def _send_code(self, code_list, protocol, pulselength):
+        """Send the code(s) with a specified pulselength."""
+        _LOGGER.info("Sending code(s): %s", code_list)
+        for code in code_list:
+            self._rfdevice.tx_code(code, protocol, pulselength)
```

Comment:
Good to know! I honestly know not enough about these kind of things to explain it, but I can tell you that the solution the way it is now works with my switch, which needs 4 different codes sent in a row. Things like removing the conditional and other things caused it to simply not switch on or off for whatever reason.

I also tried to mess with signal_repetitions, but I didn't really help, no matter what other code than the one you see here worked.
I find it quite weird, because if I send the codes with the rpi_rf component with the example code on the repo directly without going through HA, it's no issue at all. It works even with the response loop being checked every time and it works very quickly. It just seems that with HA for whatever reason it doesn't work even though according to the debug it sends all the codes the same way, maybe just with a slightly higher delay between them and I have no idea how I can prioritize the task more.